cmd/virt-chroot/tap-device-maker.go

```diff
@@ -20,6 +20,9 @@ func createTapDevice(name string, owner uint, group uint, queueNumber int) error
 		Group:      uint32(group),
 	}
 
+	if queueNumber == 1 {
```

Comment:
I saw your comments on the library and this is indeed confusing. Could we still instead of this "hack", which does what the netlink library does internally use the confusing but official API? Basically: If number is 1, set `0` on the tab device and a comment why? Open to discuss this, but I personally would understand it better than this way.

Reply:
I guess we think differently here; explicitly setting the flags is imho using the official API (it is a public member of the `TunTap` struct ... ). 

Setting `queueNumber` to 0 when I have `queueNumber = 1` is what I consider a hack .

Reply:
I don't care too much. You can also leave it but it definitely needs a comment. It is not understandable without it either way. No matter how we express it, we do this to get what the library calls `legacy` behaviour and in both cases it is not clear that this is what gives us that.

Reply:
I read this like this: If a single queue was defined, then use the default tap flags.
Previously it implicitly did the same: If no definition was set, use the default tap flags. e.g. being explicit 1 or not mentioning anything is the same.

Do I understand this wrong? If not, the only comment I can think of is to link to the bug or better, just keep the commit message with the relevant info.
Anyway, this may be a matter of taste, I have no strong feeling on this (except disliking comments and keeping distance from them as much as possible).

Reply:
> I read this like this: If a single queue was defined, then use the default tap flags.
> Previously it implicitly did the same: If no definition was set, use the default tap flags. e.g. being explicit 1 or not mentioning anything is the same.

As I read it: Create a tap device where I don't define any flags, but if there is only one queue, it should use the default flags. 

This is weird, since why would I not get the default flags if I don't specify any flags, or what is the difference now to the default flags to what I would get otherwise? Finally, if I specify now the default flag, do I maybe also miss some flags which would get implicitly added based on my requests which I would still want to have? Here a comment solves it. Something like `This forces the netlink library to create the tap device in non-multi-queue mode, since it also opens it in multiqueue mode even if just one queue is  specified, which differs from libvirt`.

Reply:
I've added a comment explaining why this is required, and linking to the bug in netlink.